### PR TITLE
Improve UX by regrouping all project-related settings within QGIS' project properties dialog and settings within QGIS' options dialog

### PR DESCRIPTION
**Pull request by nirvn**
This PR moves the QFieldSync project properties away from a disconnected dialog _into_ the QGIS project properties dialog (under upcoming QGIS 3.16 LTR):
Peek 2020-09-17 13-03

Regrouping all project-related settings into a single dialog offers a faster and more logical UX. For e.g., a user can now set read-only layers and qfieldsync layer actions within the same dialog in one go. 

The project CRS status bar button is now a nice shortcut to get to QFieldSync project properties too :+1: 

---

Same improvement has been applied to QField's settings dialog:
Screenshot from 2020-09-17 15-47-49

It feels much more integrated that way.

### CODE REVIEW

**Comment by nirvn** [comment]
@suricactus , as discussed earlier this week.

**Comment by nirvn** [comment]
Oups, forgot to mention one point in the PR initial description. By appending QFieldSync project properties into the QGIS project properties dialog, the plugin appears much more integrated into QGIS itself, which is a nice cozy feeling to deliver to users.

**Comment by nirvn** [comment]
The QField panel in the layer properties dialog now uses the high res logo used for project properties:
Screenshot from 2020-09-17 12-59-38

File: qfieldsync/gui/project_configuration_dialog.py [hunk 1]
```diff
@@ -17,239 +17,39 @@
...
+from qfieldsync.gui.project_configuration_widget import ProjectConfigurationWidget
 
-class ProjectConfigurationDialog(QDialog, DialogUi):
+class ProjectConfigurationDialog(QDialog):
     """
     Configuration dialog for QFieldSync on a particular project.
     """
 
-    def __init__(self, iface, parent=None):
+    def __init__(self, parent=None):
```
[the remark this answers is not in the document]
Comment (signedav) [reply qfieldsync/gui/project_configuration_dialog.py:33]: Or should it open there the project property with the widget as well?

[the remark this answers is not in the document]
Comment (nirvn) [reply qfieldsync/gui/project_configuration_dialog.py:33]: Fixed that in the latest commit.

File: qfieldsync/qfield_sync.py [hunk 2]
```diff
@@ -34,15 +34,48 @@
...
+from qgis.core import Qgis, QgsOfflineEditing, QgsProject
+
+from qgis.gui import (
+    QgsOptionsWidgetFactory,
+    QgsOptionsPageWidget,
+)
 
 from qfieldsync.gui.package_dialog import PackageDialog
-from qfieldsync.gui.preferences_dialog import PreferencesDialog
+from qfieldsync.gui.preferences_widget import PreferencesWidget
```
[the remark this answers is not in the document]
Comment (nirvn) [reply qfieldsync/qfield_sync.py:45]: Fixed, my bad.

**Review by signedav — APPROVED** [review]
Looks good.

**Pull request merged**